src/cast_core/src/SampleHttpClient.cpp

```diff
@@ -0,0 +1,284 @@
+            // happen if you don't stop the metrics clock (e.g. if there is an uncaught
+            // exception). See the full documentation on `OperationContext`.
+            //
+            // We don't care about how long the value-generator takes to run
+            // so we don't start the clock until after evaluating it. Purists
+            // would argue we even start until after we've done the below log
+            // statement. You decide.
+            //
+            auto inserts = _totalInserts.start();
+
+            //
+            // You have the full power of boost logging which is rendered to stdout.
+            // Convention is to log generated documents and "normal" events at the
+            // debug level.
+            //
+            BOOST_LOG_TRIVIAL(debug) << " SampleHttpClient Inserting "
+                                    << bsoncxx::to_json(document.view());
+
+            //
+            // ⚠️ If your Actor throws any uncaught exceptions, the whole Workload will
+            // attempt to end as quickly as possible. Every time this inner loop loops
+            // around, it checks for any other Actors' exceptions and will stop iterating
+            // if it sees any. Such failed workloads are considered "programmer error"
+            // and will mark the workload task as a system-failure in Evergreen. ⚠️
+            //
+
+            //
+            // Actually do the logic.
+            //
+            // We expect that this insert operation may actually fail sometimes and we
+            // wouldn't consider this to be programmer-error (or a configuration-
+            // error) so we catch try/catch with the expected exception types.
+            //
+            auto host = "google.com";
+            auto port = "443";
+            auto stream = http::make_tls_stream("google.com", "443");
+
+            // Set up an HTTP GET request message
+            boost::beast::http::request<boost::beast::http::string_body> req{boost::beast::http::verb::get, "/", 11};
+            req.set(boost::beast::http::field::host, host);
+            // Send the HTTP request to the remote host
+            boost::beast::http::write(stream, req);
+
+            // This buffer is used for reading and must be persisted
+            boost::beast::flat_buffer buffer;
+
+            // Declare a container to hold the response
+            boost::beast::http::response<boost::beast::http::dynamic_body> res;
+
+            // Receive the HTTP response
+            boost::beast::http::read(stream, buffer, res);
+
+            // Write the message to standard out
+            std::cout << res << std::endl;
+
+            // Gracefully close the stream
+            boost::beast::error_code ec;
+            stream.shutdown(ec);
+
+            try {
+
+                config->collection.insert_one(document.view());
+
+                inserts.addDocuments(1);
```

Comment:
instead of the `insert_one()` call, let's put the boost stuff here in the try block, then call `inserts.success()`. This will add a metric for each https call. The `addBytes()` and `addDocuments()` calls can be removed too since they're not applicable.

Reply:
Also you probably want to rename the `inserts` metrics context to something with http in the name.